### PR TITLE
Add option to reset leak tracking.

### PR DESCRIPTION
**Pull request by polina-c**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -260,7 +260,7 @@ for troubleshooting:
 1. By passing `stackTraceCollectionConfig`
 to `withLeakTracking` or `enableLeakTracking`.
 
-<video src="md_images/collect_callstack.mov" controls="controls" playsinline="playsinline"></video>
+https://user-images.githubusercontent.com/12115586/208321882-ecb96152-3aa7-4671-800e-f2eb8c18149e.mov
```
Comment (CoderDake) [inline README.md:263]: Does this automatically show in the markdown? or do we need to turn it into a clickable link?

File: lib/src/leak_tracker.dart [hunk 2]
```diff
@@ -24,11 +24,16 @@ ObjectTracker _theObjectTracker() {
 ///
 /// The leak tracking will function only for debug/profile/developer mode.
 /// See usage guidance at https://github.com/dart-lang/leak_tracker.
```
Comment (CoderDake) [inline lib/src/leak_tracker.dart:26]: JSDOC info should be added for the new param

File: lib/src/leak_tracker.dart [hunk 3]
```diff
@@ -24,11 +24,16 @@ ObjectTracker _theObjectTracker() {
...
-void enableLeakTracking({LeakTrackingConfiguration? config}) {
+void enableLeakTracking({
+  LeakTrackingConfiguration? config,
+  bool resetIfEnabled = false,
```
Comment (CoderDake) [inline lib/src/leak_tracker.dart:29]: This name is a bit confusing, I don't get an idea of what is accomplished when it is set.

File: lib/src/leak_tracker.dart [hunk 4]
```diff
@@ -24,11 +24,16 @@ ObjectTracker _theObjectTracker() {
...
+}) {
   assert(() {
     final theConfig = config ??= const LeakTrackingConfiguration();
-    if (_objectTracker.value != null)
-      throw StateError('Leak tracking is alredy enabled.');
+    if (_objectTracker.value != null) {
+      if (!resetIfEnabled) throw StateError('Leak tracking is alredy enabled.');
```
Comment (CoderDake) [inline lib/src/leak_tracker.dart:34]: Is there a reason why we allow both behaviours? Could we just always reset ?

File: README.md (hunk 1 above)
Comment (polina-c) [reply README.md:263]: Original way did not work, but this does. and I do not know why. See here: https://github.com/dart-lang/leak_tracker/blob/e4edef3cfd431f8c438e6353194da0eba197e118/README.md

File: lib/src/leak_tracker.dart (hunk 4 above)
Comment (polina-c) [reply lib/src/leak_tracker.dart:34]: Normally leak tracking should be set once for application. So, the second one is a smell.

Reset is needed just for tests.

File: lib/src/leak_tracker.dart (hunk 3 above)
Comment (polina-c) [reply lib/src/leak_tracker.dart:29]: Renamed to resetIfAlreadyEnabled. Does it help?

File: lib/src/leak_tracker.dart (hunk 4 above)
Comment (CoderDake) [inline lib/src/leak_tracker.dart:34]:
```suggestion
      if (!resetIfEnabled) throw StateError('Leak tracking is already enabled.');
```

**Review by CoderDake — APPROVED** [review]

**Pull request merged**